### PR TITLE
Fix subtle bugs and align with IDE store

### PR DESCRIPTION
**Pull request by HNOONa-0**
# Description
Instead of manually updating the state, use commits instead. Fixed some subtle bugs with the editor. `layout.isInitialised` was behaving strange, so I instead used a standalone boolean flag when initializing the layout.

Part of: #7639

# Comments

# Checklist:

- [x] The code follows the [coding guidelines](https://github.com/omegaup/omegaup/wiki/Coding-guidelines) of omegaUp.
- [x] The tests were executed and all of them passed.

### CODE REVIEW

**Comment by HNOONa-0** [comment]
@pabo99 I'm trying to create a problem with a custom `validator` but I get an error. Is it not possible/supported?
Screenshot from 2024-06-07 10-57-05

**Comment by pabo99** [comment]
> @pabo99 I'm trying to create a problem with a custom `validator` but I get an error. Is it not possible/supported? Screenshot from 2024-06-07 10-57-05

The error you are receiving when creating problems with a custom validator is due to the .zip file not containing the necessary elements for this type of problem. 

Specifically, when you create a problem wit `custom` validator, it is necessary to add a `validator.$lang$` file where `$lang$` indicates the language in which the validator will be created. You can find more information at this [link](https://github.com/omegaup/omegaup/wiki/Generaci%C3%B3n-Manual-de-Archivo-.zip-para-Problemas#validatorlang-opcional). 

The bad news is that it is only available in Spanish.

**Comment by pabo99** [comment]
Hmm, there is a test in Cypress whose behavior is somewhat flaky. I've tried to reproduce it in my local environment, but I haven't had success doing so.

**Comment by HNOONa-0** [comment]
> Hmm, there is a test in Cypress whose behavior is somewhat flaky. I've tried to reproduce it in my local environment, but I haven't had success doing so.

Yeah this test failed in my most recent prs.

**Review by pabo99 — APPROVED** [review]

**Review by heduenas — APPROVED** [review]

File: frontend/www/js/omegaup/grader/ephemeral.js [hunk 1]
```diff
@@ -981,7 +993,7 @@ window.addEventListener(
   },
   false,
 );
-
+// when does this piece of code run
```
Comment (heduenas) [inline frontend/www/js/omegaup/grader/ephemeral.js:996]: did you mean to delete this line?

Comment (HNOONa-0) [reply frontend/www/js/omegaup/grader/ephemeral.js]: I have not seen this piece of code run yet on embed version of editor could be worth exploring later on.

**Pull request merged**